trl/trainer/utils.py

```diff
@@ -318,31 +318,32 @@ def tokenize_batch_element(
         batch = {}
 
         if not self.is_encoder_decoder:
-            chosen_tokens = self.tokenizer(chosen, add_special_tokens=False)
-            rejected_tokens = self.tokenizer(rejected, add_special_tokens=False)
-            prompt_tokens = self.tokenizer(prompt, add_special_tokens=False)
+            # Check issues below for more details
+            #  1. https://github.com/huggingface/trl/issues/907
+            #  2. https://github.com/EleutherAI/lm-evaluation-harness/pull/531#issuecomment-1595586257
+            #  3. https://github.com/LianjiaTech/BELLE/issues/337
 
-            eos_token_id = self.tokenizer.eos_token_id
-            # Get indices in list prompt_tokens["input_ids"] that equals the EOS token (often 0)
-            eos_indices_prompt = [i for i, x in enumerate(prompt_tokens["input_ids"]) if x == eos_token_id]
-            # attention mask these indices to eos_token_id
-            new_attention_mask = [
-                0 if i in eos_indices_prompt else p for i, p in enumerate(prompt_tokens["attention_mask"])
-            ]
-            prompt_tokens["attention_mask"] = new_attention_mask
-
-            # do the same for chosen and rejected
-            eos_indices_chosen = [i for i, x in enumerate(chosen_tokens["input_ids"]) if x == eos_token_id]
-            new_attention_mask_c = [
-                0 if i in eos_indices_chosen else p for i, p in enumerate(chosen_tokens["attention_mask"])
-            ]
-            chosen_tokens["attention_mask"] = new_attention_mask_c
-
-            eos_indices_rejected = [i for i, x in enumerate(rejected_tokens["input_ids"]) if x == eos_token_id]
-            new_attention_mask_r = [
-                0 if i in eos_indices_rejected else p for i, p in enumerate(rejected_tokens["attention_mask"])
-            ]
-            rejected_tokens["attention_mask"] = new_attention_mask_r
+            assert isinstance(prompt, str)
```

Comment:
```suggestion
            if not isinstance(prompt, str):
                raise ValueError(f"prompt should be an str, got {type(prompt)}")
```
Can you do something similar for all `assert` below? 🙏